src/python/gudhi/representations/metrics.py

```diff
@@ -10,17 +10,169 @@
+        if vectorj_sum != 0:
+            vectorj = vectorj/vectorj_sum
+        return np.arccos(  min(np.dot(np.sqrt(vectori), np.sqrt(vectorj)), 1.)  )
+
+def sklearn_wrapper(metric, X, Y, **kwargs):
+    """
+    This function is a wrapper for any metric between two persistence diagrams that takes two numpy arrays of shapes (nx2) and (mx2) as arguments.
+    """
+    if Y is None:
+        def flat_metric(a, b):
+            return metric(X[int(a[0])], X[int(b[0])], **kwargs)
+    else:
+        def flat_metric(a, b):
+            return metric(X[int(a[0])], Y[int(b[0])], **kwargs)
+    return flat_metric
+
+PAIRWISE_DISTANCE_FUNCTIONS = {
+    "wasserstein": hera_wasserstein_distance,
+    "hera_wasserstein": hera_wasserstein_distance,
+    "persistence_fisher": persistence_fisher_distance,
+}
+
+def pairwise_persistence_diagram_distances(X, Y=None, metric="bottleneck", **kwargs):
+    """
+    This function computes the distance matrix between two lists of persistence diagrams given as numpy arrays of shape (nx2).
+
+    Parameters:
+        X (list of n numpy arrays of shape (numx2)): first list of persistence diagrams. 
+        Y (list of m numpy arrays of shape (numx2)): second list of persistence diagrams (optional). If None, pairwise distances are computed from the first list only.
+        metric: distance to use. It can be either a string ("sliced_wasserstein", "wasserstein", "hera_wasserstein" (Wasserstein distance computed with Hera---note that Hera is also used for the default option "wasserstein"), "pot_wasserstein" (Wasserstein distance computed with POT), "bottleneck", "persistence_fisher") or a function taking two numpy arrays of shape (nx2) and (mx2) as inputs.
+
+    Returns: 
+        numpy array of shape (nxm): distance matrix
+    """
+    XX = np.reshape(np.arange(len(X)), [-1,1])
+    YY = None if Y is None else np.reshape(np.arange(len(Y)), [-1,1]) 
+    if metric == "bottleneck":
+        try: 
+            from .. import bottleneck_distance
+            return pairwise_distances(XX, YY, metric=sklearn_wrapper(bottleneck_distance, X, Y, **kwargs))
+        except ImportError:
+            print("Gudhi built without CGAL")
+            raise
+    elif metric == "pot_wasserstein":
+        try:
+            from gudhi.wasserstein import wasserstein_distance as pot_wasserstein_distance
+            return pairwise_distances(XX, YY, metric=sklearn_wrapper(pot_wasserstein_distance,  X, Y, **kwargs))
+        except ImportError:
+            print("POT (Python Optimal Transport) is not installed. Please install POT or use metric='wasserstein' or metric='hera_wasserstein'")
+            raise
+    elif metric == "sliced_wasserstein":
+        Xproj = compute_persistence_diagram_projections(X, **kwargs)
+        Yproj = None if Y is None else compute_persistence_diagram_projections(Y, **kwargs)
+        return pairwise_distances(XX, YY, metric=sklearn_wrapper(sliced_wasserstein_distance_on_projections, Xproj, Yproj))
```

Comment:
I was thinking of a general mechanism, like PAIRWISE_DISTANCE_FUNCTIONS but giving 2 functions (projection, distance_on_projections). But if sliced Wasserstein is the only one that benefits from a projection, then that's completely unnecessary and it is indeed better to do it by hand, as you did.

Reply:
sliced Wasserstein is the only one for now so I let it as it is, but I agree it would be interesting to implement more general functions in the future if required.